lib/misc.py

```diff
@@ -33,6 +34,13 @@ def __call__(self, *args, **kwargs):
 # Argument: command line arguments Namespace. Do not need to call sys.exit()
 # because caller manages that.
 
+def delete(cli):
+   imgdir = cli.storage + '/img/' + cli.image_ref
+   if(os.path.isdir(imgdir)):
+      ch.rmtree(imgdir)
```

Comment:
Here you will want to validate that it's an image before you delete it. There's code elsewhere that does this already, so use that. It may be a function you can just call, or maybe you need to refactor until it is.

Reply:
I believe this code is correct. But, it should go in a function. I think the procedure would be something like:

1. Create a function in `charliecloud.py` called `is_image_upacked` or (better) `image_unpacked_p` ([p suffix notation](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node69.html)).
2. Put the above test in it.
3. Call that function here as well as existing places the test is used.